### PR TITLE
Kalibro client integration

### PR DESCRIPTION
**Pull request by diegoamc**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: ruby
 rvm:
-  - 2.1.2
+  - 2.1.5
```
Comment (rafamanzo) [inline .travis.yml]: We need to update here to 2.2.0

File: config/environments/production.rb [hunk 2]
```diff
@@ -72,31 +68,12 @@
...
-    authentication:       'plain',
-    enable_starttls_auto: true  }
-
-  # Exception Notification
-  config.middleware.use ExceptionNotification::Rack,
-  :email => {
-    :email_prefix => "[KalibroProcessor Error] ",
-    :sender_address => %{"mezurometrics" <mezurometrics@gmail.com>},
-    :exception_recipients => %w{mezuro-core@lists.ime.usp.br}
-  }
```
Comment (rafamanzo) [inline config/environments/production.rb]: Why are we removing the exception notification?

Comment (diegoamc) [inline config/environments/production.rb]: I don't know, I believe it was you who changed this file...

File: .travis.yml (hunk 1 above)
Comment (diegoamc) [inline .travis.yml]: This was updated on branch fixing_travis.

File: config/environments/production.rb (hunk 2 above)
Comment (rafamanzo) [inline config/environments/production.rb]: It was probably a mistake in the rush for this commit:
https://github.com/mezuro/kalibro_processor/commit/8aa46399337d8462d687a2bdefab86a2751d3c6a

I'll restore this.

Em Thu Jan 22 2015 at 15:39:16, Diego de Araújo Martinez Camarinha <
notifications@github.com> escreveu:

**Comment by rafamanzo** [comment]
Would be nice to at least one more person review this pull request since it is huge and I may have missed something!

File: lib/metric_result_aggregator.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class MetricResultAggregator
+  def self.aggregated_value(metric_result)
+    values = DescriptiveStatistics::Stats.new(metric_result.descendant_values)
+    if metric_result.value.nil? && !values.empty?
+      # Fix for the period while we're using vintage Kalibro
+      form = metric_result.metric_configuration.aggregation_form.to_s.downcase
+      form = "mean" if form == "average"
+
```
Comment (diegoamc) [inline lib/metric_result_aggregator.rb]: Does this mean we can remove these 2 previous lines?

File: lib/processor/aggregator.rb [hunk 4]
```diff
@@ -21,10 +23,12 @@ def self.aggregate(pre_order_module_results)
         already_calculated_metric_results = module_result_child.metric_results.map { |metric_result| metric_result.metric}
 
         @all_metrics.each do |metric|
-          unless already_calculated_metric_results.include?(metric)
-            metric_result = MetricResult.new(metric: metric, module_result: module_result_child, metric_configuration_id: metric_configuration(metric).id)
-            metric_result.value = metric_result.aggregated_value
-            metric_result.save
+          if module_result_child.kalibro_module.granularity > Granularity.new(metric.scope.to_s.to_sym)
+            unless already_calculated_metric_results.include?(metric) # FIXME: this probably is useless now with the above if
```
Comment (diegoamc) [inline lib/processor/aggregator.rb]: What is useless? The unless statement?

**Pull request merged**